src/libostree/ostree-repo-refs.c

```diff
@@ -1215,10 +1215,10 @@ ostree_repo_list_collection_refs (OstreeRepo                 *self,
                   continue;
                 }
 
-              if (match_collection_id != NULL && g_strcmp0 (match_collection_id, current_collection_id) != 0)
+              if (match_collection_id != NULL && g_strcmp0 (match_collection_id, remote_collection_id) != 0)
```

Comment:
Hm, so before in that case `current_collection_id` looks like it was actually an uninitialized pointer.  It seems weird to me that the tests pass both ways.  I haven't looked closely at the tests - is there something wrong there that causes this to spuriously pass before?

Reply:
I didn't try very hard, but the test reliably segfaulted for me without initializing the pointer. However, if it's uninitialized, I suppose it could point to a random address that had something strcmp could handle, even 0.

Reply:
I don't think it passed. I think the CI wasn't using `--enable-experimental-api`, and it only passed for me before I put in the last minute change to the handling of `current_collection_id`.